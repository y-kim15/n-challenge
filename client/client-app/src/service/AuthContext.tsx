import React from 'react';

export const AuthContext = React.createContext<ContextType | null>(null);

// export function useAuth() {
//     return useContext(AuthContext);
// }
