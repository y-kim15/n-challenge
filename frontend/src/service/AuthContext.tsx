import React from 'react';

const AuthContext = React.createContext<ContextType | null>(null);
export default AuthContext;
// export function useAuth() {
//     return useContext(AuthContext);
// }
