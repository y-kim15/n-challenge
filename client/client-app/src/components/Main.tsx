import React, { useContext, useState, useEffect } from 'react';
import { AuthContext } from '../service/AuthContext';
import SearchTab from './SearchTab';
import ResultsTable from './ResultsTable';
import { makeStyles } from '@material-ui/core/styles';

//import axios from 'axios';

const useStyles = makeStyles((theme) => ({
    app: {
        textAlign: 'center',
    },
    header: {
        backgroundColor: '#003399',
        minHeight: '20vh',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        fontSize: 'calc(5px + 1vmin)',
        color: 'white',
    },
    body: {
        backgroundColor: '#f3f5e4',
        color: 'black',
    },
}));

const Main: React.FC = () => {
    const { token, previousRecords } = useContext(AuthContext) as ContextType;
    const [records, setRecords] = useState<IRecord[]>(previousRecords);

    // useEffect(() => {
    //     //token === '' && login();
    //     setRecords(previousRecords);
    // });

    const handleNewSearch = (result: IRecord) => {
        setRecords({
            ...records,
            [result.url]: result.output,
        });
    };

    const classes = useStyles();

    return (
        <div className={classes.app}>
            <header className={classes.header}>
                <h1>Simple Text Counter App</h1>
            </header>
            <div className={classes.body}>
                <SearchTab handleNewSearch={handleNewSearch} />
                {records.length > 0 ? (
                    <div>
                        <h3>Your Previous Searches</h3>
                        <ResultsTable results={records} />{' '}
                    </div>
                ) : (
                    <p> Any previous requests will show here! </p>
                )}
            </div>
        </div>
    );
};

export default Main;
