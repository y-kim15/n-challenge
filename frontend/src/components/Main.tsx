import React, { useContext, useState } from 'react';
import AuthContext from '../service/AuthContext';
import SearchTab from './SearchTab';
import ResultsTable from './ResultsTable';
import { makeStyles } from '@material-ui/core/styles';
import { Container } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    app: {
        backgroundColor: '#f3f5e4',
        marginTop: theme.spacing(8),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    header: {
        backgroundColor: '#003399',
        minHeight: '10vh',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        fontSize: 'calc(5px + 1vmin)',
        color: 'white',
        marginTop: theme.spacing(8),
        padding: theme.spacing(2, 6, 2, 6),
    },
    body: {
        color: 'black',
        padding: theme.spacing(2, 6, 2, 6),
    },
    msg: {
        alignItems: 'center',
    },
}));

const Main: React.FC = () => {
    const { token, previousRecords } = useContext(AuthContext) as ContextType;
    const [records, setRecords] = useState<IRecord[]>(previousRecords);

    const handleNewSearch = (result: IRecord) => {
        setRecords({
            ...records,
            [result.url]: result.output,
        });
    };

    const classes = useStyles();

    return (
        <Container component="main" maxWidth="lg">
            <div className={classes.app}>
                <header className={classes.header}>
                    <h1>Simple Text Counter App</h1>
                </header>
                <div className={classes.body}>
                    <SearchTab handleNewSearch={handleNewSearch} />
                    {records.length > 0 ? (
                        <div id="previous">
                            <h3>Your Previous Searches</h3>
                            <ResultsTable results={records} />{' '}
                        </div>
                    ) : (
                        <p className={classes.msg}>Any previous requests will be displayed here!</p>
                    )}
                </div>
            </div>
        </Container>
    );
};

export default Main;
