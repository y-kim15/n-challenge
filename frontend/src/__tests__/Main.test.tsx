import { mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import Main from '../components/Main';
import AuthContext from '../service/AuthContext';
import 'jest-localstorage-mock';
import renderer from 'react-test-renderer';
import { TableCell, TableRow, Table, TableHead } from '@material-ui/core';

test('should display message if no previous records available', async () => {
    const wrapper = mount(
        <AuthContext.Provider
            value={{
                token: 'dummyToken',
                previousRecords: [],
            }}
        >
            <Main />
        </AuthContext.Provider>,
    );
    expect(wrapper.find('p').text()).toEqual('Any previous requests will be displayed here!');
});

test('should display table with previous records', async () => {
    const wrapper = mount(
        <AuthContext.Provider
            value={{
                token: 'dummyToken',
                previousRecords: [
                    {
                        id: 'id1',
                        url: 'dummy_url',
                        created: new Date().toLocaleDateString(),
                        output: ' dummy:1, output:1',
                    },
                ],
            }}
        >
            <Main />
        </AuthContext.Provider>,
    );
    expect(wrapper.find('h3').at(1).text()).toEqual('Your Previous Searches');

    const firstTableRow = wrapper.find(Table).find(TableHead).find(TableRow).first();
    const tableCell = firstTableRow.find(TableCell);
    expect(tableCell.find('Submitted URL')).toBeTruthy();
    expect(tableCell.find('Date Created')).toBeTruthy();
});

describe('Main snapshots', () => {
    // let mount;

    // beforeAll(() => {
    //   mount = createMount();
    // });

    // afterAll(() => {
    //   mount.cleanUp();
    // });
    test('should have previous records displayed if available on landing', async () => {
        const wrapper = renderer
            .create(
                <AuthContext.Provider
                    value={{
                        token: 'dummyToken',
                        previousRecords: [
                            {
                                id: 'id1',
                                url: 'dummy_url',
                                created: new Date().toLocaleDateString(),
                                output: ' dummy:1, output:1',
                            },
                        ],
                    }}
                >
                    <Main />
                </AuthContext.Provider>,
            )
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    test('should display new landing with no previous searches for new user', async () => {
        const wrapper = renderer
            .create(
                <AuthContext.Provider
                    value={{
                        token: 'dummyToken',
                        previousRecords: [],
                    }}
                >
                    <Main />
                </AuthContext.Provider>,
            )
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
