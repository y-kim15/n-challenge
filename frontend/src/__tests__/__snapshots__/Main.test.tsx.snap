// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshots should display new landing with no previous searches for new user 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="makeStyles-app-1"
  >
    <header
      className="makeStyles-header-2"
    >
      <h1>
        Simple Text Counter App
      </h1>
    </header>
    <div
      className="makeStyles-body-3"
    >
      <div
        className="MuiPaper-root makeStyles-padding-6 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="makeStyles-page-5"
        >
          <h3>
            Type URL here
          </h3>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-align-items-xs-flex-start"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-true MuiGrid-grid-md-true"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="url"
                  id="url-label"
                >
                  URL
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={true}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="url"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="url"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-11"
                    >
                      <span>
                        URL
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Search
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <p
        className="makeStyles-msg-4"
      >
        Any previous requests will be displayed here!
      </p>
    </div>
  </div>
</main>
`;

exports[`Main snapshots should have previous records displayed if available on landing 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="makeStyles-app-1"
  >
    <header
      className="makeStyles-header-2"
    >
      <h1>
        Simple Text Counter App
      </h1>
    </header>
    <div
      className="makeStyles-body-3"
    >
      <div
        className="MuiPaper-root makeStyles-padding-6 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="makeStyles-page-5"
        >
          <h3>
            Type URL here
          </h3>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-align-items-xs-flex-start"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-true MuiGrid-grid-md-true"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="url"
                  id="url-label"
                >
                  URL
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={true}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="url"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="url"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-11"
                    >
                      <span>
                        URL
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Search
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        id="previous"
      >
        <h3>
          Your Previous Searches
        </h3>
        <div>
          <h4>
            Record here
          </h4>
          <div
            className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="collapsible table"
              className="MuiTable-root"
              role={null}
            >
              <thead
                className="MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  />
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Submitted URL
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Date Created
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root"
                role={null}
              >
                <div
                  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <tbody
                    className="MuiTableBody-root"
                    role={null}
                  >
                    <tr
                      className="MuiTableRow-root makeStyles-root-13"
                      role={null}
                    >
                      <td
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-body"
                      >
                        <button
                          aria-label="expand row"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                              />
                            </svg>
                          </span>
                        </button>
                      </td>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-body"
                        role="cell"
                        scope="row"
                      >
                        dummy_url
                      </th>
                      <td
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      >
                        28/06/2021
                      </td>
                    </tr>
                    <tr
                      className="MuiTableRow-root"
                      role={null}
                    >
                      <td
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-body"
                        colSpan={6}
                        style={
                          Object {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          }
                        }
                      />
                    </tr>
                  </tbody>
                </div>
              </tbody>
            </table>
          </div>
        </div>
         
      </div>
    </div>
  </div>
</main>
`;
